docs/core-concepts/core-sdk/cli/extensibility.md

```diff
@@ -0,0 +1,174 @@
+# .NET Core CLI extensibility model 
+
+## Overview
+This document will cover the main ways how to extend the CLI tools. It will cover the two main ways on how to extend 
+the tools and explain the scenarios that drive each of them. It will the outline how to consume the tools as well as 
+provide short notes on how to build both types of tools. 
+
+## Two different ways to extend CLI
+The CLI tools can be extended in two main ways:
+
+1. Via NuGet packages on a per-project basis
+2. Via the system's PATH
+
+The two extensibility mechanisms outlined above are not exclusive; you can use both or just one. Which one to pick 
+depends largely on what is the goal you are trying to achieve with your extension.
```

Comment:
What are the possible goals? Should we give a hint that they will have more information below?